src/App.tsx

```diff
@@ -1,25 +1,101 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
 import './App.scss';
 import './styles/general.scss';
 import { TodoList } from './components/TodoList';
 import { CurrentUser } from './components/CurrentUser';
+import { getTodos } from './api';
 
 const App: React.FC = () => {
-  const [
-    selectedUserId,
-    // setSelectedUserId,
-  ] = useState(0);
+  const [selectedUserId, setSelectedUserId] = useState(0);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [query, setQuery] = useState('');
+
+  const changeUser = (userId: number) => setSelectedUserId(userId);
+
+  const getCurrentTodos = () => {
+    getTodos()
+      .then((currentTodosFromServer) => {
+        setTodos(currentTodosFromServer);
+      });
+  };
+
+  useEffect(() => {
+    getCurrentTodos();
+  }, []);
+
+  const selectHandler = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    if (event.target.value === 'all') {
+      getCurrentTodos();
+    }
+
+    if (event.target.value === 'complited') {
+      getTodos()
+        .then((currentTodosFromServer) => {
+          setTodos(
+            currentTodosFromServer
```

Comment:
I think, you can miss promise 32-33 lines and in setTodos put todos like `setTodos(todos.filter((todo: Todo) => todo.completed)`
Because in 18 line, you have already put todos from server to todos.
it also should be work and write a less code)